x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/PutAutoFollowPatternRequestTests.java

```diff
@@ -123,4 +123,33 @@ public void testValidate() {
         validationException = request.validate();
         assertThat(validationException, nullValue());
     }
+
+    public void testValidateName() {
```

Comment:
Would you break each of the separate test cases into a separate test? `testValidateNameComma`, `testValidateNameLeadingUnderscore`, ...

Also, I'm not seeing a test that a name can otherwise contain an underscore as long as it's not the leader character.